next/src/components/cards/EventCard.tsx

```diff
@@ -1,67 +1,83 @@
 import { Typography } from '@bratislava/component-library'
 import Image from 'next/image'
-import React, { HTMLAttributes } from 'react'
+import React, { useId } from 'react'
 
 import CardBase from '@/src/components/cards/CardBase'
-import CardContent from '@/src/components/cards/CardContent'
-import MLink from '@/src/components/common/MLink/MLink'
-import FormatEventDateRange from '@/src/components/formatting/FormatEventDateRange'
-import cn from '@/src/utils/cn'
+import { CardTitleLevel } from '@/src/components/cards/getCardTitleLevel'
+import Button from '@/src/components/common/Button/Button'
+import ImagePlaceholder from '@/src/components/common/Image/ImagePlaceholder'
+import { EventPageCardEntityFragment } from '@/src/services/graphql'
+import { formatDate } from '@/src/utils/formatDate'
+import { generateImageSizes } from '@/src/utils/generateImageSizes'
 import { getLinkProps } from '@/src/utils/getLinkProps'
 import { useTranslation } from '@/src/utils/useTranslation'
 
-type EventCardProps = {
-  title: string
-  linkHref: string
-  address?: string
-  dateFrom: string
-  dateTo: string | null
-  isLongTerm?: boolean
-  imageSrc: string
-  imageSizes?: string
-  className?: string
-} & HTMLAttributes<HTMLElement>
+type Props = {
+  eventPage: EventPageCardEntityFragment
+  cardTitleLevel?: CardTitleLevel
+}
 
 /**
- * Figma: https://www.figma.com/file/17wbd0MDQcMW9NbXl6UPs8/DS-ESBS%3A-Component-library?node-id=487-938&t=7RMKJATYwi0EYY9K-0
+ * Figma: https://www.figma.com/design/17wbd0MDQcMW9NbXl6UPs8/DS--Component-library?node-id=17952-15643&t=TnBDoFjQ0SyGzgY6-4
  */
-const EventCard = ({
-  title,
-  linkHref,
-  address,
-  dateFrom,
-  dateTo,
-  isLongTerm,
-  imageSrc,
-  imageSizes,
-  className,
-  ...rest
-}: EventCardProps) => {
+
+const EventCard = ({ eventPage, cardTitleLevel = 'h3' }: Props) => {
   const { t } = useTranslation()
+  const titleId = useId()
+
+  const imageSizes = generateImageSizes({ default: '100vw', md: '33vw' })
+
+  const { title, pageBackgroundImage, pageHeaderSections } = eventPage
+
+  // Pages have always max 1 header
+  const [header] = pageHeaderSections ?? []
+  const { date, address } = header?.__typename === 'ComponentHeaderSectionsEvent' ? header : {}
+
+  const metadata = [formatDate(date), address]
+    .filter((metadataItem) => !!metadataItem)
+    .join('  •  ')
```

Comment:
In bratislava.sk project, I'm more used to see this kind of data parsing outside of the cards, and only relevant props are passed to card one by one, not as a Strapi fragment. I know it's not a 100% rule 🤷  but I think it makes sense to stick with it.

Reply:
Discussed: Please change.